src/crimson/os/seastore/seastore.cc

```diff
@@ -205,11 +206,14 @@ seastar::future<> SeaStore::write_fsid(uuid_d new_osd_fsid)
 
 SeaStore::mkfs_ertr::future<> SeaStore::mkfs(uuid_d new_osd_fsid)
 {
-  return read_meta("mkfs_done").then([this, new_osd_fsid] (auto tuple) {
+  return read_meta("mkfs_done").then([this, new_osd_fsid] (auto tuple) mutable {
     auto [done, value] = tuple;
     if (done == 0) {
       return seastar::now();
     } else {
+      if (new_osd_fsid.is_zero()) {
+        new_osd_fsid.generate_random();
```

Comment:
Hmm, in the classical OSD this logic somehow became store-specific. In `BlueStore`:

```cpp
  void set_fsid(uuid_d u) override {
    fsid = u;
  }
  uuid_d get_fsid() override {
    return fsid;
  }
```

```cpp
int BlueStore::mkfs()
{
  // ...
  r = _read_fsid(&old_fsid);
  if (r < 0 || old_fsid.is_zero()) {
    if (fsid.is_zero()) {
      fsid.generate_random();
      dout(1) << __func__ << " generated fsid " << fsid << dendl;
    } else {
      dout(1) << __func__ << " using provided fsid " << fsid << dendl;
    }
    // we'll write it later.
  } else {
```

Reply:
yes, it is test uuid zero in store level.